### PR TITLE
PbxPlus GTM

### PR DESCRIPTION
**Pull request by 500apps-Sudhacheemala**
Updated Template

### CODE REVIEW

File: templates.tpl [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  },
+  "description": "Pbxplus enables you to build powerful and interactive forms which help in gathering data for research insight.\nhttps://infinity.500apps.com/Pbxplus",
```
Comment (venkateswarluboyapati) [inline templates.tpl]: Check content.

File: templates.tpl [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+
+___TESTS___
+
+scenarios:
+- name: Test case 1
+  code: |-
+    const mockData = {
+      // Mocked field values
+      form_key:"ZGV2LTI4MTQtMTE5Ni0xMTEz",
```
Comment (venkateswarluboyapati) [inline templates.tpl]: is this key is for pbx plus

File: templates.tpl [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    // Call runCode to run the template's code.
+    runCode(mockData);
+
+    // Verify that the tag finished successfully.
+    assertApi('gtmOnSuccess').wasCalled();
+
+
+___NOTES___
+
+Created on 7/22/2021, 4:35:06 PM
```
Comment (venkateswarluboyapati) [inline templates.tpl]: Can we change to todays date

**Comment by subramaniyamkolluri** [comment]
video

File: Metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+homepage: "https://pbxplus.com"
+documentation: ""
+versions:
+  # Latest version
+  - sha: a2d425b6d3aa2cff27a553d8f3dc9ac16acfe68e
+    changeNotes: Pbxplus GTM.
+
```
Comment (subramaniyamkolluri) [inline Metadata.yaml]: no empty lines

File: README.md [hunk 5]
```diff
@@ -1 +1,53 @@
-# 500apps-public-gtm-pbxplus
+# PBX Plus by 500apps
+PBX Plus is an advanced communications system and IVR for your business needs. Configure the call flow to allow for automatic forwarding so that employees can spend their time on building long-lasting customer relationships. Add new numbers, change call flows and use existing numbers so your customers are sure to reach you each and every time they call. Gain a better understanding of your customers and your call flow process with the ability to analyse the data and generate call reports. 
+
+Understand decisions made by customers so you can make educated business decisions based on what options the customer chooses on calls. Multi-level menus, unlimited call minutes, business hours, music-on-hold, dial by name, outbound dialer, conference rooms, forwarding, voicemail transcriptions, click to call, and much more at one place with PBX Plus.
+
+[Get Started](https://infinity.500apps.com/Pbxplus) with easy to use and configurable virtual PBX application.
```
Comment (subramaniyamkolluri) [inline README.md]: https://infinity.500apps.com/Pbxplus

Better give website rather than app url

all small case.

File: README.md [hunk 6]
```diff
@@ -1 +1,53 @@
...
+Read through the GTM Terms of Service. Click Yes to accept them and continue.
+You will get a container with a code snippet. Copy the code and paste it into every page of your site in the recommended places. Click Ok to close the code snippet box and finish setting up the GTM manager and your first container.
+
+### Installation 
+
+1. Adding the PBX Plus tag to your website once you're logged in to your account with the container, it's time to add a new tag with the PBX Plus tracking code inside it. To do so, follow these steps
+In the Google Tag Manager Home, click on the New Tag tile.
+
+
+2. Click on the Tag Configuration tile and search for our community template 'PBX Plus by 500apps' in the 'Community Template Gallery'.
```
Comment (subramaniyamkolluri) [inline README.md]: PBXPlus

File: README.md [hunk 7]
```diff
@@ -1 +1,53 @@
...
+
+
+3. Click on ‘Add to workspace’ to select the template and add it to your workspace
+After the template is added to your workspace, the tag is automatically added to your tag configuration.
+
+
+4. Find the generated PBX Plus key for the department , copy and place the key in the field provided in the Tag Configuration. You can find the PBX Plus key in the Share Script slide bar in the PBX Plus web application for each department. Just click on Google Tag Manager in the Share Script slide bar and you can find the key and the instructions.
```
Comment (subramaniyamkolluri) [inline README.md]: instructions are not step by step. Where is department

Try above steps and see it works. It will never

File: README.md [hunk 8]
```diff
@@ -1,41 +1,41 @@
...
 
-Understand decisions made by customers so you can make educated business decisions based on what options the customer chooses on calls. Multi-level menus, unlimited call minutes, business hours, music-on-hold, dial by name, outbound dialer, conference rooms, forwarding, voicemail transcriptions, click to call, and much more at one place with PBX Plus.
+Understand decisions made by customers so you can make educated business decisions based on what options the customer chooses on calls. Multi-level menus, unlimited call minutes, business hours, music-on-hold, dial by name, outbound dialer, conference rooms, forwarding, voicemail transcriptions, click to call, and much more at one place with PBXPlus.
 
-[Get Started](https://infinity.500apps.com/Pbxplus) with easy to use and configurable virtual PBX application. 
+[Get Started](pbxplus.com) with easy to use and configurable virtual PBX application.
```
Comment (subramaniyamkolluri) [inline README.md]: https://pbxplus.com

File: README.md [hunk 9]
```diff
@@ -1,41 +1,41 @@
...
+Understand decisions made by customers so you can make educated business decisions based on what options the customer chooses on calls. Multi-level menus, unlimited call minutes, business hours, music-on-hold, dial by name, outbound dialer, conference rooms, forwarding, voicemail transcriptions, click to call, and much more at one place with PBXPlus.
 
-[Get Started](https://infinity.500apps.com/Pbxplus) with easy to use and configurable virtual PBX application. 
+[Get Started](pbxplus.com) with easy to use and configurable virtual PBX application. 
 
-Automate call processes, optimize team productivity, and improve customer experience with best-in-class virtual phone system.- PBX Plus(https:/pbxplus.com)
+Automate call processes, optimize team productivity, and improve customer experience with best-in-class virtual phone system.- PBXPlus(pbxplus.com)
```
Comment (subramaniyamkolluri) [inline README.md]: same

File: README.md [hunk 10]
```diff
@@ -1,41 +1,41 @@
...
-2. Click on the Tag Configuration tile and search for our community template 'PBX Plus by 500apps' in the 'Community Template Gallery'.
+2. Click on the Tag Configuration tile and search for our community template 'PBXPlus by 500apps' in the 'Community Template Gallery'.
 
 
 3. Click on ‘Add to workspace’ to select the template and add it to your workspace
 After the template is added to your workspace, the tag is automatically added to your tag configuration.
 
 
-4. Find the generated PBX Plus key for the department , copy and place the key in the field provided in the Tag Configuration. You can find the PBX Plus key in the Share Script slide bar in the PBX Plus web application for each department. Just click on Google Tag Manager in the Share Script slide bar and you can find the key and the instructions.
+4. Find the individually generated PBXPlus key in each call flow, copy and place the key in the field provided in the Tag Configuration. You can find the PBXPlus key in the Share Script slide bar in the PBXPlus web application for each call flow. Just click on Google Tag Manager in the Share Script slide bar and you can find the key and the instructions.
```
Comment (subramaniyamkolluri) [inline README.md]: not good

File: Metadata.yaml [hunk 11]
```diff
@@ -0,0 +1,6 @@
+homepage: "https://pbxplus.com"
+documentation: ""
+versions:
+  # Latest version
+  - sha: a2d425b6d3aa2cff27a553d8f3dc9ac16acfe68e
+    changeNotes: Pbxplus GTM.
```
Comment (subramaniyamkolluri) [inline Metadata.yaml]: PBXPlus

Its a brand name keep it consistent.

File: templates.tpl [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+  },
+  "description": "PBX Plus is an advanced communications system and IVR for your business needs.\nhttps://infinity.500apps.com/ PBXPlus",
```
Comment (subramaniyamkolluri) [inline templates.tpl]: PBXPlus

Comment (subramaniyamkolluri) [reply templates.tpl]: why there is a space
Screenshot 2021-12-08 at 8 05 36 AM

File: templates.tpl [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+    "type": "TEXT",
+    "name": "api_key",
+    "displayName": " PBXPlus Key",
+    "simpleValueType": true,
+    "valueValidators": [
+      {
+        "type": "NON_EMPTY"
+      }
+    ],
+    "help": "Please copy the  PBXPlus from App -\u003eCreate a form -\u003e Embed. https://infinity.500apps.com/ PBXPlus"
```
Comment (subramaniyamkolluri) [inline templates.tpl]: same

File: templates.tpl [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+]
+
+
+___SANDBOXED_JS_FOR_WEB_TEMPLATE___
+
+const log = require('logToConsole');
+const injectScript = require('injectScript');
+const encodeUriComponent = require('encodeUriComponent');
+const api_key = encodeUriComponent(data.api_key);
+const trackingUrl='https://gtm.500apps.com/gtm/ PBXPlus/'+api_key+'.js';
```
Comment (subramaniyamkolluri) [inline templates.tpl]: why there is space


Screenshot 2021-12-08 at 8 06 35 AM
no formatter

**Comment by subramaniyamkolluri** [comment]
@500apps-Sudhacheemala Can these be fixed?

**Comment by 500apps-Sudhacheemala** [comment]
> @500apps-Sudhacheemala Can these be fixed?

Yes sir

File: templates.tpl (hunk 14 above)
Comment (500apps-Sudhacheemala) [reply templates.tpl]: resolved

File: templates.tpl (hunk 13 above)
Comment (500apps-Sudhacheemala) [reply templates.tpl]: resolved

File: templates.tpl (hunk 12 above)
Comment (500apps-Sudhacheemala) [reply templates.tpl]: resolved

File: templates.tpl [hunk 15]
```diff
@@ -42,7 +42,7 @@ ___TEMPLATE_PARAMETERS___
         "type": "NON_EMPTY"
       }
     ],
-    "help": "Please copy the  PBXPlus from App -\u003eCreate a form -\u003e Embed. https://infinity.500apps.com/ PBXPlus"
+    "help": "Please copy the  PBXPlus from App -\u003eCreate a form -\u003e Embed. https://infinity.500apps.com/PBXPlus"
```
Comment (subramaniyamkolluri) [inline templates.tpl]: urls are always small case. 

https://infinity.500apps.com/pbxplus

fix other places

Comment (500apps-Sudhacheemala) [reply templates.tpl]: resolved

**Pull request merged**